### DISCUSSION TITLE
[Feature] Facial Recognition Enhancements

### DISCUSSION THREAD

**Discussion by Dental1347** [immich-app/immich#2472] [General]
# (Parent discussion for FR enhancements)

Summary:

- [x] 1. Ability to show/hide faces (similar to google photos) (#3262)
- [x] 2. Option to select an alternative person thumbnail (similar to google photos) (#3081)
- [ ] 3. Explore having a person "preview" when view people in a shared album, link, partner timeline, etc.
- [ ] 4a. Improve performance of viewing all people 
- [x] 4b. Improve performance of viewing all assets for a person (#3557)
- [x] 5. Add the option to link/merge two people into one (#3121)
- [x] 6. Add a confirmation modal before running recognize faces for all assets (#2496)
- [ ] 7. Add FR related features into the mobile app
- [x] 8. Possibility of a dedicated `People` sidebar item (#4257)
- [x] 9. Explore reducing required clicks to tag people (#3328)
- [x] 10. Add the option to remove/update a wrongly tagged person (#4949)
- [x] 11. Back arrow is broken when navigating to a person from the `View All` (people) page. (#2566)
- [ ] 12. Run the FR pipeline again for a specific image.

### The feature

Hi,

Awesome that the FR is implemented, it works awesome! After making use of the UI if felt a bit clunky and could perhaps use a bit of work. Down below I have listed a few things that I've noticed.

1. Currently within the UI there are 12 spots, which I presume are reserved for people with the most pictures. Although, I doubt this a bit. There are various things that came to mind when I saw this. It would be a fantastic feature to favorite certain people, this could be put under the Explore tab. At the same time adding a separate tab for Facial recognition, or People, could also be a possibility. 
3. It would be nice if there is a possibility to add more than 12 people within the `people` tag within explore. 
![image](https://github.com/immich-app/immich/assets/44779473/4eab57f3-8c02-4ce3-81b5-1689f57f282d)
10. When you want to assign a name to someone that is not within the Explore tab you have to click on `view all`, then click on the person, and then you're able to add the name. A few things came to mind, add the possibility to add names to the image it self, as per shown below. Second
![image](https://github.com/immich-app/immich/assets/44779473/b00dd736-edb6-47e0-8c5a-5c10e4a3a0c7)
12. When clicking on a photo, the faces that are within the photo are displayed. I don't know if this is possible or even feasible, but it would be nice if the faces are displayed that are actually within the picture, instead of the general image which is associated with the person. This can help with improper matching, although I have not yet encountered this.

If there is more, I will edit it below.

**EDIT:
13. When clicking on `view all` and selecting a person, after clicking on the back arrow in the top left. It takes you back to the explore page, resulting in needing to click on `view all again`. When clicking on back arrow within the browser, it does not do this. Although, I have also noticed this not working 100% of the time. This just adds another click.
14. The possibility to remove faces or hide them.
15. The possibility to unmark someone if they are incorrectly tagged.
16. A rescan on a per image basis if facial recognition does not recognize a particular face in the image. (I've noticed this happening with people who have a mouctache)**

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
Thanks for documenting these!

**Comment by colin715** [comment]
I would appreciate a separate page/menu icon for People

**Reply by jrasm91** [reply]
We're pretty much sticking to the layout that Google uses right now, although we could certainly consider alternatives.

**Reply by dominikheinz** [reply]
I agree, a dedicated tab to manage everything related to people and FR would be nice.

**Reply by jrasm91** [reply]
An option to enable a dedicated "People" page link (number 8 in the list) has been added and will be in the next release.

**Comment by dominikheinz** [comment]
Great list. Another feature that came to mind after experimenting with the FR feature is the option to "link" people. For example, I have two pictures of myself, clearly in frame, but they are detected as different people. (Although to the human eye they look very similar).

So, it would be great if I could merge these "two" people into one, or link them.

I feel like a dedicated section in the settings for everything related to people and face recognition would be a good idea.

**Reply by Belphemur** [reply]
Link or merge them. I think they should be one "person".
One easy way I saw it done was to merge the people when we use the exact same name.

**Reply by bjrne** [reply]
I'd prefer same names to not be merged automatically, otherwise you have to keep in mind who you tagged already if you're not using family names...

**Reply by bread22** [reply]
"merge" is a great feature to have, my son shows up as 3 different people...

**Comment by ShinasShaji** [comment]
Another feature I can suggest is having the ability to set a 'Key' photo of a person that would then be the image displayed in most of the UI except where `4` above comes into play. This is possible in Apple's photos app as 'Make key photo', and in Google Photos as 'Use as feature photo'. 

I think this would be great because in the current UI it is often possible for blurry or yawed faces to be key photos.

**Comment by DrSpaldo** [comment]
Some of the ideas for tweaking the UI are good ones. Having to do the view all, then click the photo, then name it, then back to the main list and start again is a little cumbersome

**Comment by Vcele** [comment]
In Google Photos there's also a way to improve recognition by manually assigning select photos to a person. Also, like in Photoprism, single faces could be moved into their own overview to prevent clutter.

**Comment by DrSpaldo** [comment]
I've just found my first person that was incorrectly linked to a people/person tag. Ie. There is 300 photos that have been tagged/linked to a particular person, however, one is incorrect and not that person. Is there a way I can unlink that photo from that person? If there is no way, can that please be added to the roadmap above?

**Reply by jrasm91** [reply]
Added!

**Comment by jeremywillans** [comment]
Is it possible to highlight in a photo where each identified face is? Scenario is when FR detects partial faces, if its possible to locate this in a photo in order to work out who it actually is to link to an existing user, or correctly identify them.

Also, the option to dismiss people/hide from the people list

**Reply by fabiannagel** [reply]
There's now a [discussion](https://github.com/immich-app/immich/discussions/3489) for this

**Comment by Dental1347** [comment]
I've added two new suggestions. To my post, as I intended the first list to be edited by the team :)

**Comment by xyxzxyz** [comment]
This is an excellent list. However, any chance you can move the additional requests above "(Original Post Below)"? I almost overlooked the #12...#15. I scrolled past Original Post and missed those additional requests.

**Reply by Dental1347** [reply]
You want me to move them to the list above the post?

**Comment by Harris-KA** [comment]
My thoughts exactly! 5 would be great for me as anyone with sunglasses seems to be a different person!

**Comment by pablo03v** [comment]
5 please :)

**Comment by henley-regatta** [comment]
I'm finding the "All People" view very slow to load, because FR has found >2000 unique faces; that's a lot of thumbnails to load.
Would it be possible to add a "Named People" and "Unknown People" (without tags) filters to the interface to help with this in everyday use?

**Reply by DrSpaldo** [reply]
I can't remember if it was in one of the Github discussions or in Discord. But I know this issue has been raised and was on the list of things to look at. I believe in part 'lazy loading' may assist with this as well..

**Reply by henley-regatta** [reply]
Thanks. 
I've since discovered that the bulk of my own performance issue was due to a conflict between the way this page is being built and a Chrome extension I'm using (HoverZoom+). Time to first thumbnail with that active was >10 seconds; if I disable it it's ~1.5seconds. 
(I'm writing this here mainly in case anyone else is having this sort of performance issue)

**Reply by DrSpaldo** [reply]
Very interesting @henley-regatta , I should take a look at what extensions I have running as well to see if it would help my terrible loading times and partial freezing

**Reply by petersuba** [reply]
After running FR through my archive and ending up with 10000+ unique faces, most of them are small blurred faces of people in a crowd, I also think it would be useful to be able to specifically tag faces as "Unclear" or something similar, specifically to mean that a.) yes, these are correctly identified as faces (I don't want to drop their association with a face so that FR wouldn't come back offering them up again next time), b.) they are unusable for practical purposes and I don't care about them (similar but not the same as "hidden": a person may be recognised and his/her faces may be grouped but then the specific person would be hidden - in this case this is a collection of many faces that would likely be many persons also). 

In terms of UI/UX:
- it would be useful to "tag" faces as this from multiple places, e.g. 
  - when in "Edit faces" for a single photo, this option can be next to the "select new face" button
  - in the People page, this option can be added to the "hamburger" menu popping up for a face after "Hide person"
  - In the page for a single face's pictures, when selecting multiple photos, in can be on the context menu next to "Fix incorrect match" and/or as another button on the "Fix incorrect match" page next to "Create new person"
- faces with this "tag" should be treated in terms of their visibility the same way as those with "Hide person", except when using Show/hide hidden faces options, instead of just a list of hidden faces, these pictures could be shown as a separate category/grouping of photos (this is really just icing on the cake for fixing erroneous clicks)

**Reply by alextran1502** [reply]
@petersuba you can tune the number of similar faces to create a person in the machine learning settings. Default you need at least 3 similar faces to create a person. By increasing it to like 5-6 I think it will reduce the number of people get created significantly

**Reply by petersuba** [reply]
@alextran1502 I am aware of that, thank you. Even if I do that, I have many photos in from events where there is a crowd sitting in an audience and so there ends up being many pictures in a "person" that isn't real. My library overall has 200000+ photos over 2 decades. There is only so much one can do tweaking ML parameters as well before the valuable matches start to get dropped as well, not to mention the time it takes to run the model. If at least there was a way to re-run the model with keeping the named people/faces only and re-running the rest (an outstanding feature request) that would help losing the additional manual work, but even then running the model again and again takes time and processing power.

**Comment by radh21301** [comment]
Would it be possible to implement a feature to detect faces in videos? I know Google Photos does some magic and shows people in videos as well (not just people found in the video thumbnail). This is could be such a cool feature!

**Reply by pablo03v** [reply]
If the person is visible in the thumbnail it detects it already, "at least"

**Reply by Ziomal12** [reply]
It detects though. I even checked right now to make sure and surely when I click on one of the faces in first 3 lines of photos I've got 2 videos.

**Reply by Ziomal12** [reply]
Anyways, if anyone wants to play around with face detection in a video I'd suggest to analyze 1 frame per second of video or even less (maybe even 1 per 5 seconds)

**Reply by radh21301** [reply]
I checked again. I can see that people are detected if they are present in the video thumbnail (or in the very beginning of the video). But not when they come in the middle of the video. Maybe I am doing something wrong.

**Reply by xyxzxyz** [reply]
It does recognize faces in videos, at least in mine.

**Reply by tsemachh** [reply]
Perhaps consider allow having in the person view in video links to the positions this person appears , this would allow finding people on specific location on the video

**Comment by lukanowa** [comment]
I have some suggestions to hide/show faces ability:
1. Merging peoples option should not show hidden faces,
2. When we want to hide more faces all faces with no name on it should show on top below named faces (it hard to find missing faces to hide when we have over 300 faces)
3. When we try to name faces and this name is already use can Immich ask to merge this two people?

**Reply by 0rn0lf** [reply]
Totally agree to point 1.
Additionally to point 2, I think it would make more sense to be able to select multiple faces and hide them all at once whitin the faces page and not the hidden faces. Could look like this:
![image](https://github.com/immich-app/immich/assets/35098710/463f8f18-daf3-4bf9-9344-3771872ee893)
Otherwise you would always have to scroll trough all recognized faces, hidden or not. 

Also, the visibility is not good to me. Sometimes its hard to see which face is not yet hidden when scrolling through the list. Maybe a small outline highlight to non-hidden faces.
![image](https://github.com/immich-app/immich/assets/35098710/eed44f14-f6ff-413d-b61b-f6da165a8c4e)

But the biggest improvement would be merging multiple faces at one. This kinda connects to point 1.
I dont know if this is technically possible but if not, it shouldnt be a big deal to merge one face after another from the selected ones as background task by just selecting a destination face and the faces you want to merge into this.

**Reply by martabal** [reply]
> Additionally to point 2, I think it would make more sense to be able to select multiple faces and hide them all at once whitin the faces page and not the hidden faces. Could look like this:

At this point, just use the merge editor ?

**Reply by 0rn0lf** [reply]
merge editor? Where can I find it?

**Reply by martabal** [reply]
The "Merge Faces" button.

**Reply by yoyotueur** [reply]
Would it be possible to add an autocomplete when faces have not been recognised and you type for the name so you make sure you don't input the wrong format for existing people

**Reply by martabal** [reply]
Yes, this is in my TODO list

**Comment by hassan-sd** [comment]
Do we have any ETA's for the remainder? ie #10 is something I'm facing currently as an issue across 8,000 images where the AI is detecting the young face of my son as if it's my daughter when she was a baby. So a lot of photos are incorrectly tagged as her when in face it's him.

To suggest, it would be good to allow batch selecting of images and update faces, not just one at a time. 

What I'd like do is go to a date range and pretty much select all the baby photos and re-tag them as another person to batch edit them

**Reply by DrSpaldo** [reply]
I’m the same, it would be great to exclude 3 years for one of the children as they weren’t born yet!

**Reply by Soulplayer** [reply]
This would be really awesome. I’m currently seeing one child but it contains 4 children. The oldest is now 9 and the youngest 1 so a filter between dates would work great.

**Reply by jrasm91** [reply]
We could maybe do something like that, but there is no way to use that data for future matches without processing the images themselves again. So basically if you want to auto match future images we have to reprocess them. Also, we can't use the image as a profile picture without the bounding box data for the face. Regardless it could be interesting to see if we could leverage the data for improving the initial clustering algorithm.

**Comment by eygraber** [comment]
Google auto completes names when adding a label to a face. It would be nice if immich could do that as well so I don't have to type out the whole name and remember exactly what I named that face.

**Comment by el-tiuri** [comment]
Another great addition would be the ability to extract face data written by programs like digiKam and Picasa. I have a lot of photos with embedded face data, and it feels like a waste to have to re-tag all photos the moment you want to use a different library manager. This is something that [Pigallery2](https://bpatrik.github.io/pigallery2/) supports, to name an example.

I don’t know if it would be possible to combine it with Immich’s own face recognition (perhaps by using it as a way to pre-check found faces if direct conversion to its format isn’t possible?)

**Reply by AngelaDMerkel** [reply]
Google Photos and Lightroom can also embed face data into images, but some of it is erroneous

**Comment by shakyr** [comment]
The features I'd like regarding facial recognition/people:
- Ability to tag someone as in an image, disconnected from the facial recognition. I have some photos where people have their back to the camera or they're too young, but I'd still like to say they're in the photo.
- This would also be useful for tagging pets.
- Ability to blacklist photos or albums from the facial recognition. Especially if you have photos of crowded areas and a few dozen random  people show up as recognised.
- Bonus points if you can blacklist all unmatched faces in a photo (so you can tag known people first and then ignore everyone else).

**Reply by AngelaDMerkel** [reply]
Google Photos absolutely has some implementation of your first feature and it is exceptionally useful. I would like to +1 for a face whitelist. Only tagging faces of those manually approved would be a good feature, especially when you may have several images of a crowded area where faces repeat, or where you have a acquaintance often in your images, but not significant enough for you to tag the face.

**Reply by Judman** [reply]
Yes, So many unknown and blurry background faces just muck up this feature.

**Reply by jhit** [reply]
Yes I already missed that feature a lot. :+1:

**Reply by mraction** [reply]
It would be so nice to have this feature!

**Reply by whoindahouse** [reply]
It would also be nice to have the option to feed a face you tag back to the facial recognition model for improvement.

**Reply by l-mb** [reply]
Can this be added to the list of items above or as a separate FR?

I can live with imperfect facial recognition AI (it'll never be perfect), but that I can't manually tag people as being in a picture greatly reduces the ability to find all pictures from the "explore" tab.

There's tags, but the workflow is not great and it is disconnected from the "People" section. Maybe a special `People/...` tag hierarchy and show all images tagged as that person's name alongside that person in Explore?

**Reply by amadeo-alex** [reply]
+1 should be a separate PR

**Reply by dkipkaev** [reply]
I just wanted to tag the cat, I agree

**Reply by arcstur** [reply]
I would also really love this. I've been creating manual albums, but then it splits the logic between faces and albums. And yes, we want to tag the cat as well! (in my case dog)

**Reply by kmsigma** [reply]
I'm +1-ing this because Immich's facial recognition is only recognizing one of my dogs.  Yes, I'm serious.

**Comment by jakob42** [comment]
Maybe we could add #4096 to the list, might be better here than in its own feature request.

**Comment by r3df0chs** [comment]
For Documentation my report https://github.com/immich-app/immich/issues/4194
When i load the people page, i think all images will be loaded from the web server, because of that limiting factors of a proxy for example can kick in. It would be nice when the thumbnails will be loaded when the part of the page loads, instead of all thumbs

**Comment by AuraHerreroRuiz** [comment]
I would like to add, even if I'm not sure how much effort it would take, to have face recognition for pets! I would guess mainly cats and dogs, as that's what Google Photos is capable of doing, although I haven't searched if they tag other animals.

**Reply by lionhoho** [reply]
I had my dog recognized, but only few photos, although I have hundreds photos for my dog. Don't know why.

**Reply by Vyerni** [reply]
I have the very same situation. My dog entered as a person in about 4 photos, but have about 500.

**Reply by ridicolos** [reply]
Would love to see pet support too. I got two dogs with more than enough photos of both of them. Only one dog was recognized with just two photos of it

**Reply by 0rn0lf** [reply]
At least your dog wasn’t recognised as a sheep 😂
But I also have a ton of animal photos and only one of them was recognised. 
Also, it must be smart enough so that it doesn’t add every animal in a zoo to your people list.

**Reply by mertalev** [reply]
I'd love to add this, but there doesn't seem to be much to work with in terms of models or even training data.

You can get something similar with smart search by searching for your pet's breed / features. Combined with smart albums (once we add that), this can simulate pet detection to some extent.

**Reply by sym0nd0** [reply]
Excuse my n00b question. Would it be possible to train the local model from a selection of uploaded photos, which the user manually tags and then when a scheduled scan runs or after future uploads any that match automatically get tagged, with a yes/no vote to improve the local model?

As I've no clue what I'm talking about, I also have no idea if that's possible, a stupid idea or how difficult it would be to implement.

**Reply by Chuckame** [reply]
Go hitting the up-vote button in this discussion to show interest about pets and make it prioritised -> https://github.com/immich-app/immich/discussions/7151

**Reply by radh21301** [reply]
The devs don't prioritize by the number of upvotes. But sure, I just gave it a thumbs up :)

**Comment by aviv926** [comment]
Number 10 is done

**Comment by theshoeshiner** [comment]
Can manual tagging (https://github.com/immich-app/immich/discussions/2472#discussioncomment-6850486) be added to the summary list? I think it's been mentioned a few different times but I don't see it called out as a separate requested feature and I don't think it falls under any of the current ones.

**Comment by Soulplayer** [comment]
Probably really difficult to implement but what I like about apple photos app is that it also suggests faces to match with a person (e.g. is this person person y?). Is this something that could be added to the list?

**Comment by xyxzxyz** [comment]
Auto add person/pet to album is a must.

**Comment by thomasloven** [comment]
Some form of "mass retagging" would be awesome too.
I have a person with 200+ assets where perhaps 100 are wrong but of the same second person.
Right now it's six clicks to retag each.

I wish I had some idea on how a better way to do it might even look...

**Reply by tippfehlr** [reply]
select all wrong faces and click on the three dots -> ‘fix incorrect match’ :)

**Reply by jakob42** [reply]
I cannot find that on v1.94.1. Where did you find that? I've got a couple pictures selected and want to change one (wrongly) recognized face into another...

**Reply by tippfehlr** [reply]
Sorry, I wasn't specific. You have to select them on the page for a face.

[search] -> people -> click on the face the images are wrongly assigned to -> select images -> click the three dots at the top right

**Reply by thomasloven** [reply]
I have no idea how I could have missed that...

Thanks a lot!

**Comment by tippfehlr** [comment]
~~I have two people shown as one (because of similar faces) a few times.~~

~~Currently, there is no way of manually adding faces (?), so the wrong face can’t be changed manually.~~
~~So, first step would be to allow manually creating faces~~
~~But manually making FR more specific in these cases (by telling FR which face is which) would be great.~~

EDIT: I just found the (much more powerful) menu when clicking on ‘fix incorrect match’ in the page for a face.

**Comment by geoleven** [comment]
One great feature would be to add the ability for facial detection / recognition to be able to run on all assets without losing the previous tags.

Let me explain:
Right now there are two options, missing and all. Missing runs for images it has not already run, all will remove all the tags (for me that I have more than 600 named people, this is a week's worth of work). Right now, and depending on some conditions (bugged out facial recognition, etc which I probably have to file as a bug at some point), I end up in a lot of situations with obvious missing faces detected and only the already named people are tagged. I would like to have a middle option to rerun the face detection on the parts of the photographs that is not tagged already. I don't really know the technical background so I am not sure this is technically easy or viable. Another take would probably be (if the previous one is undoable), to enable a per photo - on demand rerun of the facial detection. This means that although I may lose the info for the already tagged people for a photo is easy to reassign this specific one adding the new ones.

**Reply by tuupola** [reply]
Something like this is needed. It is also currently bit too easy to loose all your named persons.

**Comment by jenda69** [comment]
It would be nice to have (or is it there somewhere?) option to rescan face in a photo, similarly to "Refresh metadata" and "Refresh thumbnails" commands.
I have an external gallery where I manage my photos externally, that especially means editing RAWs. And when I'm done editing, I run a script to refresh edited output JPG photos in Immich using API command `runAssetJobs` with `regenerate-thumbnail` and `refresh-metadata` parameters. I get new thumbnails etc. but if I cropped a photo, the faces are now tagged in wrong positions when hovering over them in details panel.
It doesn't really break anything, but I find it a bit annoying.

**Reply by waclaw66** [reply]
Really missing such feature. Currently using SQL workaround...
```sql
delete from asset_faces where "assetId"='DESIRED_ASSET_ID'
```
Following `Refresh thumbnails` recognizes asset faces again. With caution!

**Comment by jhit** [comment]
I would love to see an option to share FR Data with a partner account. It is a huge effort to maintain and correct recognized faces. I want to share the training data and already tagged persons with my family. Furthermore, I want my family to be able to see and search by people on  my shared pictures.

**Reply by aviv926** [reply]
https://github.com/immich-app/immich/pull/7045

**Reply by tuupola** [reply]
I am also looking for this feature. We have a group of people recognizing people from old photos. Currently this is possible only by using one shared account, which is not optimal.

**Comment by isaacolsen94** [comment]
Hey all, I've seen manually being able to tag faces mentioned a few times, but I don't see it on the list of features...is this something the team is looking into? Or is there no plan in implement it?

**Reply by alextran1502** [reply]
it is something we want to do eventually for sure

**Reply by isaacolsen94** [reply]
Awesome! That's good to hear! We know there is a lot of stuff on your plate, thanks for confirming!

**Reply by tuupola** [reply]
Manually tagging a person to a face would be great. For example if I have four photos of a person, three of them are recognized and person is created. Currently there is no easy way to add the fourth photo to the person.

Only way I can make this work is to set the `MIN RECOGNIZED FACES` faces to 1 and then keep merging the single image persons. Or maybe I am missing something?

**Reply by MacBadger** [reply]
Manual tagging is my number one! It would be great. The only feature Immich doesn't have!

But just to highlight at the top, "**remove**/update a wrongly tagged person" is ticked as complete. This isn't the case as far as i can tell. You can reassign. If say a none-person (or stranger!) is detected as being someone, you can't remove the tag... only change it to someone else.

**Reply by lukaslindnermusic** [reply]
I also would absolutely love this feature! This would improve the app so much 😍

**Comment by whoindahouse** [comment]
Any way to increase the number of unnamed faces you can merge into a named face from 5? Even temporarily...very useful when first scanning a library

**Reply by filski** [reply]
I haven't tried yet, but: https://github.com/immich-app/immich/discussions/3894#discussioncomment-7923497

**Comment by porjo** [comment]
After adding a bunch of photos and now have a lot faces to sort through. Some things that could make this easier:
- merge faces by dragging a face thumbnail onto another face thumbnail (saves three clicks)
- double clicking on the face thumbnail would cause the 'change name' form to appear (saves one click)

Thankyou to everyone involved in creating Immich!!

**Reply by whoindahouse** [reply]
Or the ability to select multiple faces from the people page via CTRL+click and merge them into one

**Reply by porjo** [reply]
@whoindahouse yes I think CTRL+click might be better. Dragging and drop could get annoying if it becomes necessary to scroll the page up/down to find the target thumbnail.

**Comment by radh21301** [comment]
**Detect 'person' using clothing and other visual cues across photos taken within a similar timeframe --- Like Google Photos**

I this this is a super cool feature in google photos. I was sure surprised when I saw the results the first time. Would be really nice to do something like this in Immich as well!

https://www.androidauthority.com/google-photos-facial-recognition-back-3333108/


Edit: Looks like there is another feature request with the same improvements for facial recognition (https://github.com/immich-app/immich/discussions/9810)

**Comment by themistymay** [comment]
In reference to this request: https://github.com/immich-app/immich/discussions/3356
Unmerging faces (incorrectly detecting multiple kids as the same person). Match was done by FR not merge feature so this would not be a reverse of merge operation, but rather a retrain / rerun after pulling out bad matches.

The issue is that no other "person" is the other kid so there is nothing to reassign and it seems that adding a person is not correctly allowing a FR workflow.

**Comment by Germs2004** [comment]
Please allow the user to "confirm" an AI-recognized face so Immich will know it's recognized correctly.  Immich should also have a view of all the unconfirmed faces, per person, so we can highlight a bunch at a time to mark them confirmed.  If Immich ever rescans all the photos to update its guesses about who each face belongs to, it should keep all the user-confirmed faces and only edit the guesses of unconfirmed faces.

**Comment by Germs2004** [comment]
If I set a person's Date of Birth in Immich, the AI facial recognition should consider that information along with the photo's Date Photo Taken metadata when deciding which person that face belongs to.  It should not recognize a face of an adult as being a person who has a date of birth 3 years ago.

**Comment by Germs2004** [comment]
If Immich assigns a face in a photo to the wrong person, and I edit that face to the correct person, I'd like a way to tell Immich to rescan all the faces with that new information I gave it so it will automatically correct any other mistakes that are similar to this one.  This feature seems to need the "user-confirmed face" feature I mentioned above so it will know which faces are definitely correct and which ones are its own previous guesses.  It should adjust its remaining unconfirmed guesses based on the photos I've since confirmed.

For example, I may have 1000 photos of John and 1000 of Sally, but 20 photos of Sally are incorrectly marked as John, so when I edit one of those mistakes from John to Sally, Immich should fix the other 19 automatically.

**Comment by bo0tzz** [comment]
I'm going to close this thread since most of the original points have been implemented and it's impossible to track dozens of feature requests in one thread like this. If there's a request in here that you want to see and that hasn't been implemented yet, please _check for duplicates_ and then open a new FR if there isn't one yet.
